src/sql/parts/notebook/notebook.component.ts

```diff
@@ -479,4 +483,16 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		}
 	}
 
+	public async runAllCells(): Promise<boolean> {
+		await this.modelReady;
+		let codeCells = this._model.cells.filter(cell => cell.cellType === CellTypes.Code);
+		for (let i = 0; i < codeCells.length; i++) {
```

Comment:
codeCell will be undefined if there is no code cell exists in Notebook, therefore, codeCells.length results in exception. You may want to safeguard it ?